src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2044,10 +2044,10 @@ private BoundExpression BindSuppressNullableWarningExpression(PostfixUnaryExpres
             var type = expr.Type;
             if ((object)type != null)
             {
-                // Report diagnostic if the type consists of no reference types.
+                // Report diagnostic if the type is a value type.
                 // PROTOTYPE(NullableReferenceTypes): Rather than checking for reference types,
                 // check whether the suppression is necessary (in NullableWalker) and warn if not.
-                if ((object)type.VisitType((t, a, c) => t.IsErrorType() || !t.IsValueType, (object)null, canDigThroughNullable: true) == null)
+                if (!type.IsErrorType() && type.IsValueType)
```

Comment:
Consider merging with outer `if`.

Reply:
Done. I've also removed the `IsErrorType` check, since error types aren't value types (I've sealed `ErrorTypeSymbol.IsValueType` to make that unambiguous).

FYI, I've also rebased the PR to resolve conflicts (that added some LangVersion diagnostics).